activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -109,6 +109,10 @@ def supports_optimizer_hints?
         !mariadb? && database_version >= "5.7.7"
       end
 
+      def optimizer_hint_allowed_in_subquery?(hint)
+        !hint.include?("MAX_EXECUTION_TIME")
```

Comment:
From what I could find (https://dev.mysql.com/doc/refman/5.7/en/server-error-reference.html#error_er_warn_unsupported_max_execution_time) this is the only hint that's supported only on the top level